### ISSUE TITLE
Style/HashAsLastArrayItem copy with style "no_braces" can result in invalid code

### ISSUE THREAD

**Issue by wcmonty** [rubocop-hq/rubocop#8475]
## Expected behavior

Using Style/HashAsLastArrayItem with EnforcedStyle: no_braces should yield valid Ruby code when there are duplicate Hashes in the array.  This is similar to https://github.com/rubocop-hq/rubocop/issues/8323, but has not been fully resolved when there are identical hashes in the array. 

## Actual behavior

The cop is not safe and results in invalid Ruby code.

## Steps to reproduce the problem
```yaml
# .rubocop.yml
AllCops:
  NewCops: enable

Style/HashAsLastArrayItem:
  EnforcedStyle: no_braces
```

```ruby
# hash_as_last_array_item.rb before
# frozen-string-literal: true

# Note that the hashes are identical
a = [
  {
    foo: :bar,
  },
  {
    foo: :bar
  }
]

pp a
```

```console
$ ruby hash_as_last_array_item.rb
[{:foo=>:bar}, {:foo=>:bar}]
```

```console
$ rubocop -a hash_as_last_array_item.rb
Inspecting 1 file
E

Offenses:

hash_as_last_array_item.rb:5:3: C: [Corrected] Style/HashAsLastArrayItem: Omit the braces around the hash.
  { ...
  ^
hash_as_last_array_item.rb:6:14: C: [Corrected] Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.
    foo: :bar,
             ^
hash_as_last_array_item.rb:7:3: E: Lint/Syntax: unexpected token tCOMMA
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  ,
  ^
hash_as_last_array_item.rb:8:3: C: [Corrected] Style/HashAsLastArrayItem: Omit the braces around the hash.
  { ...
  ^

1 file inspected, 4 offenses detected, 3 offenses corrected
```

```ruby
# hash_as_last_array_item.rb after
# frozen-string-literal: true

a = [

    foo: :bar
  ,

    foo: :bar

]

pp a
```

```console
$ ruby hash_as_last_array_item.rb
hash_as_last_array_item.rb:7: syntax error, unexpected ',', expecting ']'
hash_as_last_array_item.rb:11: syntax error, unexpected ']', expecting end-of-input
```

## RuboCop version

```
$ rubocop -v
0.89.0
```
